src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -181,14 +181,22 @@ BoundExpression rewriteAssignment(BoundExpression leftRead)
             // assume that was the case.
 
             // If the property is static or if the receiver is of kind "Base" or "this", then we can just generate prop = prop + value
-            if (receiverOpt == null || propertyOrEvent.IsStatic || !CanChangeValueBetweenReads(receiverOpt))
+            if (receiverOpt == null || propertyOrEvent.IsStatic)
             {
                 return receiverOpt;
             }
 
             Debug.Assert(receiverOpt.Kind != BoundKind.TypeExpression);
 
             BoundExpression rewrittenReceiver = VisitExpression(receiverOpt);
+            var adjustedRewrittenReceiver = AdjustReceiverForExtensionsIfNeeded(rewrittenReceiver, propertyOrEvent, stores, ref temps);
+
+            if (!CanChangeValueBetweenReads(receiverOpt) && adjustedRewrittenReceiver == rewrittenReceiver)
+            {
+                return receiverOpt;
```

Comment:
>return receiverOpt; [](http://example.com/codeflow?start=16&length=19)

I see that you are trying to preserve what the original code was doing, but whatever it was doing looks suspicious. Why would we want to conditionally not lower a receiver? Does it mean that somewhere else we will make another attempt to lower it? Also, it looks like other call sites of ```CanChangeValueBetweenReads``` are actually using lowered node as the argument. So, I suggest trying the following logic instead:
```
            if (!CanChangeValueBetweenReads(rewrittenReceiver) && adjustedRewrittenReceiver == rewrittenReceiver)
            {
                return rewrittenReceiver;
            }
```